src/cli/check.rs

```diff
@@ -132,3 +141,373 @@ impl ExecuteSubcommand for Command {
+        self.data = None;
+        self
+    }
+
+    /// Set the data to be checked and remove potential text field.
+    #[must_use]
+    pub fn with_data(mut self, data: CliData) -> Self {
+        self.data = Some(data);
+        self.text = None;
+        self
+    }
+
+    /// Set the data (obtained from string) to be checked and remove potential
+    /// text field
+    pub fn with_data_str(self, data: &str) -> serde_json::Result<Self> {
+        serde_json::from_str(data).map(|data| self.with_data(data))
+    }
+
+    /// Set the language of the text / data.
+    #[must_use]
+    pub fn with_language(mut self, language: String) -> Self {
+        self.language = language;
+        self
+    }
+
+    /// Return the text within the request.
+    ///
+    /// # Errors
+    ///
+    /// If both `self.text` and `self.data` are [`None`].
+    /// If any data annotation does not contain text or markup.
+    pub fn try_get_text(&self) -> Result<String> {
+        if let Some(ref text) = self.text {
+            Ok(text.clone())
+        } else if let Some(ref data) = self.data {
+            match data.annotation.len() {
+                0 => Ok(Default::default()),
+                1 => data.annotation[0].try_get_text(),
+                _ => {
+                    let mut text = String::new();
+
+                    for da in data.annotation.iter() {
+                        text.push_str(&da.try_get_text()?);
+                    }
+
+                    Ok(text)
+                },
+            }
+        } else {
+            Err(Error::InvalidRequest(
+                "missing either text or data field".to_string(),
+            ))
+        }
+    }
+
+    /// Return a copy of the text within the request.
+    /// Call [`Request::try_get_text`] but panic on error.
+    ///
+    /// # Panics
+    ///
+    /// If both `self.text` and `self.data` are [`None`].
+    /// If any data annotation does not contain text or markup.
+    #[must_use]
+    pub fn get_text(&self) -> String {
+        self.try_get_text().unwrap()
+    }
+
+    /// Split this request into multiple, using [`split_len`] function to split
+    /// text.
+    ///
+    /// # Errors
+    ///
+    /// If `self.text` is none.
+    pub fn try_split(mut self, n: usize, pat: &str) -> Result<Vec<Request>> {
+        let text = mem::take(&mut self.text)
+            .ok_or_else(|| Error::InvalidRequest("missing text field".to_string()))?;
+
+        Ok(split_len(&text, n, pat)
+            .iter()
+            .map(|text_fragment| self.clone().with_text(text_fragment.to_string()).into())
+            .collect())
+    }
+
+    /// Split this request into multiple, using [`split_len`] function to split
+    /// text.
+    /// Call [`Request::try_split`] but panic on error.
+    ///
+    /// # Panics
+    ///
+    /// If `self.text` is none.
+    #[must_use]
+    pub fn split(self, n: usize, pat: &str) -> Vec<Request> {
+        self.try_split(n, pat).unwrap()
+    }
+}
```

Comment:
I think all of this should be only implemented for `Request`, not `CliRequest`. You can simply have the CLI have the following logic:

```
cli_req = CliRequest::parse();
req: Request = cli_req.into();
req.some_function...
```

Reply:
Alright, makes sense, I'll see if that works